api/notd/store/retriever.py

```diff
@@ -206,3 +210,15 @@ async def list_user_interactions(self, fieldFilters: Optional[Sequence[FieldFilt
         result = await self.database.execute(query=query, connection=connection)
         userInteractions = [user_interaction_from_row(row) for row in result]
         return userInteractions
+
+    async def list_latest_updates(self, fieldFilters: Optional[Sequence[FieldFilter]] = None, orders: Optional[Sequence[Order]] = None, limit: Optional[int] = None, connection: Optional[DatabaseConnection] = None) -> Sequence[LatestUpdate]:
```

Comment:
you should add another function similar to `get_token_metadata_by_registry_address_token_id` which retrieved just one and throws and error if it doesnt exist.

Reply:
leave this function as well tho, just add the other one and use it in the collection code below instead